### ISSUE TITLE
Bitcoin core UI is unresponsive while syncing

### ISSUE THREAD

**Issue by steevithak** [bitcoin-core/gui#804]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behaviour

I'm running Fedora with a Wayland/Gnome desktop. Each time I start Bitcoin Core, it is completely unresponsive during the initial syncing period and continuously displays "org.bitcoin.bitcoin-qt is not responding" dialogs with Force quit / Wait options. If I keep clicking the Wait option, it will eventually finish syncing and work normally. It's acting like the UI is not able to respond to normal UI events while the backend is syncing but no other Gnome applications I use behave this way.

![Screenshot from 2024-03-09 16-26-05](https://github.com/bitcoin-core/gui/assets/660193/91195c6e-4879-48a8-a2d8-0e2931652390)

### Expected behaviour

GUI app should be fully responsive to foreground UI events such as mouse clicks and keyboard actions even when the backend is busy syncing. 

### Steps to reproduce

1.Start Bitcoin Core on a Fedora Wayland/Gnome desktop and try to use it while it's syncing
2. See endless "not responding" dialogs

### How did you obtain Bitcoin Core

Package manager

### What version of Bitcoin Core are you using?

26.0-1

### Operating system and version

Fedora 38

### Machine specifications

Intel NUC 13 Pro NUC13VYKi7 ( 64GB RAM / 1TB SSD )
Fedora 38
Linux kernel 6.7.7-100
Gnome 44.9
bitcoin-core-desktop.x86_64   26.0-1.fc38

**Comment by hebasto** [comment]
> ### How did you obtain Bitcoin Core
> 
> Package manager

Which one?

**Comment by steevithak** [comment]
dnf is the package manager for Fedora. Installed from the Fedora repo via "dnf install bitcoin-core-desktop"

**Comment by hebasto** [comment]
Cannot reproduce it on my Fedora 38 with the `bitcoin-core-desktop-26.0-1.fc38.x86_64` installed.

Any additional hints that might help to reproduce the issue?

Do Guix binaries from https://bitcoincore.org/bin/bitcoin-core-26.0/ work for you?

**Comment by hebasto** [comment]
Are any wallets loaded? If so, could you try to run with `-nowallet` option?

**Comment by steevithak** [comment]
Maybe a mouse or keyboard action is required to trigger the process? What I usually do after starting the UI is try to go to the Window -> Console menu to open the console so I can add my local bitcoin node (so it'll sync over local ethernet). When I first hit the Window menu is when I first notice the "not responding" dialog popup. Each time I hit wait, it will go away for a few seconds and if I'm quick and I perform an action or two before it pops up again. I'll investigate the other binaries when I get a chance. I'm not sure I want to try installing those over the Fedora binaries on my desktop box for fear of blowing up my wallet or something but I may be be able to try them out on another box.

**Comment by steevithak** [comment]
Yes, I keep my wallet on this box and periodically load up bitcoin core to keep it synced up or make a transaction. I'm not sure how to use the --nowallet option with the desktop UI version  of bitcoin core but I'll investigate that when I get a chance.

**Comment by hebasto** [comment]
> I'm not sure how to use the --nowallet option with the desktop UI version of bitcoin core but I'll investigate that when I get a chance.

One way is to run in GNOME Terminal:
```
bitcoin-qt -nowallet
```

No wallets will be loaded automatically. This can help with your issue until full sync. You can load your wallet manually via the menu: File -> Open Wallet.

**Comment by steevithak** [comment]
> > I'm not sure how to use the --nowallet option with the desktop UI version of bitcoin core but I'll investigate that when I get a chance.
> 
> One way is to run in GNOME Terminal:
> 
> ```
> bitcoin-qt -nowallet
> ```
> 
> No wallets will be loaded automatically. This can help with your issue until full sync. You can load your wallet manually via the menu: File -> Open Wallet.

That worked to start the UI but it didn't seem to find my existing config and wants to build an entirely new blockchain DB. I'll need to figure out how to point it at my existing data. But, in any case, it's synced now so I can't reproduce the problem anymore. When it's fully synced, it starts right up and the UI works fine. Once it's a week or two out of date, I'll try it again and see if starting from the command line helps. If no one else is seeing this, feel free to close. Maybe it's something weird about Fedora 38 or my configuration. I need to update to Fedora 39 (or 40 soon) anyway...
